akka-cluster-sharding/src/main/scala/akka/cluster/sharding/ShardRegion.scala

```diff
@@ -746,10 +746,50 @@ private[akka] class ShardRegion(
 
   var coordinator: Option[ActorRef] = None
 
+  def reRegisterIfCoordinatorNotUp(): Unit =
+    if (coordinator.nonEmpty) {
+      val coordAddress = coordinator.get.path.address // safe: guarded by nonEmpty
```

Comment:
there is a difference of addresses for local actor selections, so this should be:
```
val coordAddress = {
        val a = coordinator.get.path.address // safe: guarded by nonEmpty
        if (a.hasLocalScope) cluster.selfMember.address
        else a
      }
```